### DISCUSSION TITLE
Issue and PR timelineItems query never returns CommentDeletedEvent's

### DISCUSSION THREAD

**Discussion by jamesdh** [community/community#24540] [Apps, API and Webhooks]
<p>The GraphQL endpoint allows querying for <code>timelineItems</code> on issues/PR’s. One of the listed timeline item types is a <a href="https://developer.github.com/v4/object/commentdeletedevent/" rel="nofollow noopener">CommentDeletedEvent</a>. Unfortunately, when creating/deleting a comment on an issue/PR, this event item never appears in the timeline.</p>
<p>Is this event improperly named (and intended for some other purpose)? Is this a bug? Or is this for some obscure reason by design? </p>

**Comment by parithiban** [comment, marked as answer]
<p><a class="mention" href="/jamesdh">@jamesdh</a></p>
<p>I can get this event with the following scenario.</p>
<ul>
<li>Created a pull request as User1</li>
<li>Made a comment on the pull request as User2</li>
<li>Deleted the comment as the User1 who created the pull request</li>
</ul>
<p>PR link: <a href="https://github.com/parithiban/jenkins-git-integration/pull/15" rel="nofollow noopener">https://github.com/parithiban/jenkins-git-integration/pull/15</a></p>
<p>GraphQL Query:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/001371fb768a9ce3793cb097e694feed021bfbb3.png" data-download-href="/uploads/short-url/FEZOnC8UadE9JKtBk8qW4puTN.png?dl=1" title="Screen_Shot_2020_05_23_at_11_07_32_PM" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/7281166/181085218-f6a33575-34f9-4cae-b3ec-ade3d1f5851b.png" alt="Screen_Shot_2020_05_23_at_11_07_32_PM" data-base62-sha1="FEZOnC8UadE9JKtBk8qW4puTN" width="690" height="325" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/0/001371fb768a9ce3793cb097e694feed021bfbb3_2_690x325.png, https://aws1.discourse-cdn.com/github/optimized/2X/0/001371fb768a9ce3793cb097e694feed021bfbb3_2_1035x487.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/0/001371fb768a9ce3793cb097e694feed021bfbb3_2_1380x650.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/001371fb768a9ce3793cb097e694feed021bfbb3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen_Shot_2020_05_23_at_11_07_32_PM</span><span class="informations">2872×1354 351 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by jamesdh** [comment]
<p><a class="mention" href="/parithiban">@parithiban</a> ahh so the event only exists when a comment is deleted by a user other than the original commenting user.  Thanks for pointing this out!</p>
<p>That’s inconsistent with the webhook but it kind of makes sense. The webhook still delivers an <code>issue_comment.deleted</code> event when deleting my own comment. :frowning: However if I create a comment then delete it, there doesn’t appear to be any way to know that via GraphQL.</p>
